addons/stock/models/stock_quant.py

```diff
@@ -126,11 +126,14 @@ def _compute_inventory_date(self):
         for quant in quants:
             quant.inventory_date = date_by_location[quant.location_id]
 
-    @api.depends('inventory_quantity')
+    @api.depends('inventory_quantity', 'inventory_quantity_set')
     def _compute_inventory_diff_quantity(self):
-        for quant in self:
+        has_inventory = self.filtered(lambda q: q.inventory_quantity_set)
```

Comment:
```suggestion
        has_inventory = self.filtered(lambda q: q.inventory_quantity_set or q.inventory_quantity)
```
This may avoid some inconsistencies with quants created with an `inventory_quantity` already set (like in the test you modified).

Reply:
That's so much better! Thanks!